crates/nostr-lmdb/src/store/ingester.rs

```diff
@@ -244,4 +247,25 @@ impl Ingester {
 
         Ok(false)
     }
+
+    fn handle_request_to_vanish(&self, txn: &mut RwTxn, event: &Event) -> nostr::Result<(), Error> {
+        // Acquire read txn
+        let read_txn = self.db.read_txn()?;
+
+        // Delete author events
+        self.db
+            .delete(&read_txn, txn, Filter::new().author(event.pubkey))?;
```

Comment:
I would use also the `until` condition, to delete only the events before the vanish request timestamp.